crates/proof-of-sql/src/base/database/group_by_util_test.rs

```diff
@@ -289,7 +420,39 @@ fn we_can_sum_aggregate_slice_by_counts_with_empty_result() {
 }
 
 #[test]
-fn we_can_sum_aggregate_slice_by_counts() {
+fn we_can_sum_aggregate_slice_by_counts_with_all_empty_groups() {
+    let slice_a = &[
+        100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115,
+    ];
+    let indexes = &[];
+    let counts = &[0, 0, 0];
+    let expected = &[Curve25519Scalar::from(0); 3];
+    let alloc = Bump::new();
+    let result: &[Curve25519Scalar] =
+        sum_aggregate_slice_by_index_counts(&alloc, slice_a, counts, indexes);
+    assert_eq!(result, expected);
+}
+
+#[test]
+fn we_can_sum_aggregate_slice_by_counts_with_some_empty_group() {
+    let slice_a = &[
+        100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115,
+    ];
+    let indexes = &[12, 11, 1, 10, 2, 3, 4];
+    let counts = &[3, 4, 0];
+    let expected = &[
+        Curve25519Scalar::from(112 + 111 + 101),
+        Curve25519Scalar::from(110 + 102 + 103 + 104),
+        Curve25519Scalar::from(0),
+    ];
+    let alloc = Bump::new();
+    let result: &[Curve25519Scalar] =
+        sum_aggregate_slice_by_index_counts(&alloc, slice_a, counts, indexes);
+    assert_eq!(result, expected);
+}
```

Comment:
These should never actually occur since empty groups would not be included in the result.
The tests are ok since they are testing the function, but in a sense this could be undefined behavior of the function.